agent/xds/listeners.go

```diff
@@ -1254,7 +1293,10 @@ func (s *ResourceGenerator) makeUpstreamFilterChainForDiscoveryChain(
 	if overrideCluster != "" {
 		useRDS = false
 		clusterName = overrideCluster
-		filterName = overrideCluster
+
+		if destination == "" {
+			filterName = overrideCluster
+		}
```

Comment:
What is this change here for?

Reply:
If the filterName uses the overrideCluster we end up with a filterName that contains the full SNI. That filterName is then used in the statPrefix, which should only have `name.ns.dc`.

This is a hack to get it to use the one from L1287, which is what's expected in the filter's statPrefix. 

This gets back to this function needing to be rewritten because it is _very_ confusing. I'm going to do that before the RC release.